### PR TITLE
what happens when you search google.com in your web browser

### PR DESCRIPTION
**Pull request by ayyesu**

### CODE REVIEW

File: what_happens_when_you_search_google.com_in_your_browser
Comment (Chynet35) [inline what_happens_when_you_search_google.com_in_your_browser:1]: In summary, when you type "https://www.google.com/" in your browser and press Enter, your computer goes on an exciting adventure to find Google's website, passes through guards, unlocks a secret box, gets the puzzle pieces, puts them together, and shows you the Google homepage on your screen!

**Comment by 1923k** [comment]
**### What happens when you type google.com in your browser and press Enter**

**_Please follow the below steps for the entire process_**

**No.1**
When typing https://www.google.com/ on the web browser, the browser performs a Domain Name System(DNS) lookup to find the IP address associated with “google.com”.
 
What is a DNS?

In accordance to AWS, DNS translates human readable domain names(for example, [www.amazon.com](http://www.amazon.com/) ) to machine readable address(for example, 192.0.2.44).
Every device in the internet like smartphone, laptop, desktop has its own unique number that is used to identify it and this number is called IP address. So the communication of all the device is facilitated via the IP addresses. Since, the internet has a very huge of devices connected, it is not easy for humans to remember or store all this IP addresses in their memory for every communication that can be made.
The solution to this matter is the DNS where by a human can type the domain name of the server (google.com) which still be correct as it is mapped to its IP address. So this involves sending request to DNS server for translating the domain name into an IP address.

**No.2** 
Once the client browser has obtained the IP address of the server hosting “google.com”, it will then initiate a Transmission Control Protocol(TCP) connection to that server.

**No.3**
At this stage client browser will now sent DNS request to the web server for requesting the home page of the domain name (google.com).

**No.4**
Upon receiving of the DNS request by a web server, it will then return back a HTTPS response containing the homepage of the domain (google.com) and all with the associated information required. Please note that for the security of the connection between the client and server browser, the requests and response from both would be either secured or insecure. Secure connection would have (HTTPS) to show the presence of the SSL certificate while the insecure would have (HTTP) to show the absence of SSL certificate and the connection is not trust.

**No.5**
Now the client browser will show homepage of the domain (google.com).

**No.6**
Lastly, now since a client browser has displayed the homepage, depending on what activity done by a user, the web server would the authenticate the user to the database to verify so that the required action can be done.

In all this scenario, in a case when there is too much traffic populated the web server, the would be a standby server which would automatically share the traffic with exiting server avoid overload of existing server as to improve the efficiency. This is called **Load-balancing**.

Please refer to my article here: https://www.linkedin.com/pulse/what-happens-when-you-type-googlecom-your-browser-press-kefuoe-sole/

**Comment by AnnahNokutheza** [comment]
Data will start to transmit from one computers, network, servers and routers. Once the data has been received, a DNS query will be sent from your computer to DNS Server used by the URL you are trying to access to ask for the IP address associated with the domain for you as the user to receive the necessary information contained in that "domain".

**Comment by Abe-Tshofu** [comment]
Understanding the different components involved in website delivery is crucial for web developers, network administrators, and anyone interested in the workings of the internet. Knowing how each component works together allows for better website performance, reliability, and security. 

If you're interested in learning more about the components involved in website delivery, here are some steps you can take:

Start with the basics: Begin by learning about the fundamental concepts and technologies involved in website delivery. This includes understanding how the internet works, the basics of HTML, CSS, and JavaScript, and how web browsers interact with web servers.

Study networking and protocols: Learn about networking and the various protocols used to transmit data over the internet. This includes TCP/IP, DNS, HTTP/HTTPS, and SSL/TLS. Understanding these protocols is crucial to understanding how data is transmitted between web servers and clients.

Study server-side technologies: To gain a deeper understanding of web server and application server technologies, you'll need to study programming languages, such as Java, Python, or PHP, that are commonly used to build web applications. You'll also need to learn about database systems and SQL, which are used to store and retrieve data.

Practice building websites and web applications: The best way to learn is to practice building websites and web applications. Start with small projects and work your way up to more complex projects. You can use online resources, such as tutorials and documentation, to help guide you.

Stay up-to-date with the latest technologies: The technology landscape is constantly changing, and it's important to stay up-to-date with the latest trends and developments. Follow industry blogs and attend conferences and meetups to stay informed about new technologies and best practices.

Seek guidance and mentorship: Learning on your own can be challenging, and it's helpful to have guidance and mentorship. Consider enrolling in online courses, joining online communities or forums, or finding a mentor who can help guide you on your learning journey.

Overall, learning about the components involved in website delivery requires a combination of self-study, hands-on practice, and seeking guidance from experts in the field. With dedication and persistence, anyone can learn these concepts and technologies and build their skills in web development and networking.

**Pull request closed without merging**